pkg/providers/instancetype/instancetype.go

```diff
@@ -311,6 +335,52 @@ func (p *DefaultProvider) createOfferings(ctx context.Context, instanceType *ec2
 			}).Set(price)
 		}
 	}
+
+	log.FromContext(ctx).WithValues("instanceType", *instanceType.InstanceType, "capacityReservations", capacityReservations).V(0).Info("capacity reservations for instanceType")
+	for _, capacityReservation := range capacityReservations {
+		log.FromContext(ctx).WithValues("instanceType", *instanceType.InstanceType, "capacityReservation", capacityReservation.ID).V(0).Info("capacity reservation for instanceType")
+		if capacityReservation.InstanceType != *instanceType.InstanceType {
+			log.FromContext(ctx).WithValues("instanceType", *instanceType.InstanceType, "capacityReservation", capacityReservation.ID).V(0).Info("capacity reservation not applicatble for instanceType")
+			continue
+		}
+		log.FromContext(ctx).WithValues("instanceType", *instanceType.InstanceType, "capacityReservation", capacityReservation.ID).V(0).Info("capacity reservation applicatble for instanceType")
+		price := 0.0
+		isAvailable := capacityReservation.AvailableInstanceCount > 0
+		zone := capacityReservation.AvailabilityZone
```

Comment:
nit: I think you may be able to inline some of this where you are creating the offering struct